staging/src/k8s.io/component-base/metrics/testutil/metrics_test.go

```diff
@@ -265,3 +265,80 @@ func TestHistogramValidate(t *testing.T) {
 		}
 	}
 }
+
+func TestLabelsMatch(t *testing.T) {
+	cases := []struct {
+		name          string
+		metric        *dto.Metric
+		labelFilter   map[string]string
+		expectedMatch bool
+	}{
+		{"metric's labels and labelFilter have the same labels and values", &dto.Metric{
+			Label: []*dto.LabelPair{
+				{Name: pointer.StringPtr("a"), Value: pointer.StringPtr("1")},
+				{Name: pointer.StringPtr("b"), Value: pointer.StringPtr("2")},
+				{Name: pointer.StringPtr("c"), Value: pointer.StringPtr("3")},
+			}}, map[string]string{
+			"a": "1",
+			"b": "2",
+			"c": "3",
+		}, true},
+		{"metric's labels contain all labelFilter labels, and labelFilter is a subset of metric's labels", &dto.Metric{
+			Label: []*dto.LabelPair{
+				{Name: pointer.StringPtr("a"), Value: pointer.StringPtr("1")},
+				{Name: pointer.StringPtr("b"), Value: pointer.StringPtr("2")},
+				{Name: pointer.StringPtr("c"), Value: pointer.StringPtr("3")},
+			}}, map[string]string{
+			"a": "1",
+			"b": "2",
+		}, true},
+		{"metric's labels don't have all labelFilter labels and value", &dto.Metric{
+			Label: []*dto.LabelPair{
+				{Name: pointer.StringPtr("a"), Value: pointer.StringPtr("1")},
+				{Name: pointer.StringPtr("b"), Value: pointer.StringPtr("2")},
+			}}, map[string]string{
+			"a": "1",
+			"b": "2",
+			"c": "3",
+		}, false},
+		{"The intersection of metric's labels and labelFilter labels is empty", &dto.Metric{
+			Label: []*dto.LabelPair{
+				{Name: pointer.StringPtr("aa"), Value: pointer.StringPtr("11")},
+				{Name: pointer.StringPtr("bb"), Value: pointer.StringPtr("22")},
+				{Name: pointer.StringPtr("cc"), Value: pointer.StringPtr("33")},
+			}}, map[string]string{
+			"a": "1",
+			"b": "2",
+			"c": "3",
+		}, false},
+		{"metric's labels have the same labels names but different values with labelFilter labels and value", &dto.Metric{
+			Label: []*dto.LabelPair{
+				{Name: pointer.StringPtr("a"), Value: pointer.StringPtr("1")},
+				{Name: pointer.StringPtr("b"), Value: pointer.StringPtr("2")},
+				{Name: pointer.StringPtr("c"), Value: pointer.StringPtr("3")},
+			}}, map[string]string{
+			"a": "11",
+			"b": "2",
+			"c": "3",
+		}, false},
+		{"metric's labels contain label name but different values with labelFilter labels and value", &dto.Metric{
+			Label: []*dto.LabelPair{
+				{Name: pointer.StringPtr("a"), Value: pointer.StringPtr("1")},
+				{Name: pointer.StringPtr("b"), Value: pointer.StringPtr("2")},
+				{Name: pointer.StringPtr("c"), Value: pointer.StringPtr("33")},
+				{Name: pointer.StringPtr("d"), Value: pointer.StringPtr("4")},
+			}}, map[string]string{
+			"a": "1",
+			"b": "2",
+			"c": "3",
```

Comment:
Do you mind adding two edge cases? 

(1) empty LabelPair and a non-empty map
(2) non-empty LabelPair but an empty map

Reply:
sure

Reply:
updated

Reply:
/cc @logicalhan
is it ok now?